assembl/alembic/versions/7bf35c9261d9_delete_extra_fields_for_deleted_users.py

```diff
@@ -0,0 +1,36 @@
+"""delete_extra_fields_for_deleted_users
+
+Revision ID: 7bf35c9261d9
+Revises: e3f681bd70e0
+Create Date: 2018-06-13 14:14:41.037265
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '7bf35c9261d9'
+down_revision = 'e757aefa55e1'
+
+from alembic import context, op
+import sqlalchemy as sa
+import transaction
+
+
+from assembl.lib import config
+
+
+def upgrade(pyramid_env):
+    from assembl import models as m
+    db = m.get_session_maker()()
+    with transaction.manager:
+	# SQL Alchemy can't delete after a join operation, so we are getting the ids first
+	# and deleting the corresponding profile fields
+        ids=db.query(m.ProfileField.id).join(m.User).filter(m.User.is_deleted==True).all()
+        ids = [id for (id,) in ids]
```

Comment:
```
ids_query=db.query(m.ProfileField.id).join(m.User).filter(m.User.is_deleted==True)
for (id, _) in ids_query:
   profile = m.ProfileField.get(id)
   db.delete(profile)
```

Reply:
You don't need to check for existence of the profile, as you're getting the ids right from the source (the db). Just issue a delete statement on the object itself. And you're good to go. Sad that the delete on join doesn't work :(